cli/archive.go

```diff
@@ -116,3 +117,121 @@ func addToArchive(tw *tar.Writer, filename string) error {
 
 	return nil
 }
+
+func unpackArchive(fromFile, toDir string) error {
```

Comment:
This is taken directly from `cloud_package_manager`. If we decide to keep part2 (the dylib checks), I will clean this up and refactor it into shared code (possibly in a diff PR but will do.)